### ISSUE TITLE
Imports not being found in notebook after changing the kernel

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pylance-release#5172]
Follow the directions here:
- https://github.com/bokeh/tutorial

Open the first notebook
Select a kernel that comes from the `bk-tutorial` environment

Pylance doesn't find the first import:

![image](https://github.com/microsoft/pylance-release/assets/19672699/5c2eedd7-cb39-4201-a541-c059c866f989)

If you reload VS code it will work.

Output log looks like the notebook change event didn't prompt a request for new settings:

```
2023-11-29 13:25:15.704 [info] (14140) [BG(2)] getSemanticTokens full at C:\Users\rchiodo\source\tutorial\notebooks\01_introduction.ipynb#W3sZmlsZQ==.py (3ms)
2023-11-29 13:25:23.233 [info] (Client) Jupyter provided interpreter path override: c:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\python.exe

```

That was the only thing that ended up in the log after changing the kernel.

To reproduce after reloading:
- pick a kernel that doesn't have bokeh installed
- save the notebook
- reopen VS code
- show that the import is not resolved
- change the kernel to the bk-tutorial kernel again
- the import will still not be resolved (even though the right kernel is picked)

**Comment by judej** [comment]
need to repro on stable to see if this is a regression

**Comment by rchiodo** [comment]
Okay I wasn't crazy, this seems to be caused by VS code not sending a kernel change notification:
https://github.com/microsoft/vscode/issues/199800

**Comment by rchiodo** [comment]
I can only repro it with that bokeh tutorial though.

**Comment by debonte** [comment]
@rchiodo, it looks like the upstream issue has been closed. Can this issue be closed as well?

**Comment by rchiodo** [comment]
Yeah I think this is fixed now with our latest release and the Jupyter extension's latest release.
